### ISSUE TITLE
Create trigger queue / subscription automatically

### ISSUE THREAD

**Issue by SeanFeldman** [Azure/azure-functions-host#8442]
Today, a function cannot be triggered if there’s no queue / subscription to listen to. And there’s no way to automatically provision those. Which requires the following hack: https://weblogs.asp.net/sfeldman/automatic-queue-creation-for-function. With some reflection (follow up post) it can be simplified but it would be great to have this feature ootb.

**Comment by alrod** [comment]
We do not create azure resources for triggers(storage, event hub, service bus, etc.) by design.
For example a customer can unintentionally created new resources and think a function is working without any issues.

**Comment by SeanFeldman** [comment]
Storage or Service Bus queue existence will not determine wherever a function is running or not. If a function doesn't have a single log, it cannot be running. I'm also not asking to make it the default. Rather, opt-in option to remain backwards compatible. Customers have different preferences. Some prefer to have the infrastructure rolled out in a controlled manner and use ARM (as an example) to achieve that. Others prefer to deploy a function and have the infrastructure provisioned at that time as well. I have seen with my customers this division and while the first camp gets what they need OOTB, the second camp gets nothing. Hence the feature request.

**Comment by SeanFeldman** [comment]
Also, @alrod, customers can [hack](https://weblogs.asp.net/sfeldman/automatic-queue-creation-for-function) their way through this but would rather have an option.

**Comment by stimms** [comment]
I'd agree with allowing people the option to create queues automatically. Perhaps an argument on the binding like 

```
[ServiceBusTrigger("myqueue", Connection = "ServiceBusConnection", ProvisionAutomatically = true)] 
```

Anything that reduces friction for people starting out with functions is a good plan in my mind.

**Comment by thdotnet** [comment]
To me the creation of the queue should be handled by Infrastructure (ideally IaC) during the release process, and then, deploy the function with the proper queue to listen to. For development purposes thise can be really helpful, but I'd rather do it using scripts. (my 2 cents)

**Comment by andrekiba** [comment]
I’m agree with @SeanFeldman the possibility to opt-in would be perfect 👍🏻

**Comment by SeanFeldman** [comment]
@shreyabatra4, did you close this accidentally?

**Comment by shreyabatra4** [comment]
Currently, we are not creating Azure resources for any of the triggers. This is so that we can prevent customers from accidentally creating a new resource. Transferring to the host team to investigate this feature request further.

**Comment by SeanFeldman** [comment]
> Currently, we are not creating Azure resources for any of the triggers. This is so that we can prevent customers from accidentally creating a new resource. Transferring to the host team to investigate this feature request further.

I would agree that this cautious approach when creating a resource impacts billing. Queues/subscriptions with Azure Service Bus or Storage Queues do not incur charges. There's a potential to create resources that do not follow a consistent naming convention and cause reach the quota limits, but that would be caught in the lower environments before they get into production. Therefore, it would be nice to hear the concern associated with creating resources.
